docs/ocm-integration.md

```diff
@@ -0,0 +1,249 @@
+# OCM Integration
+
+Assisted-service uses the OCM client in order to make AI clusters shown as part of all your clusters in OCM UI.
+
+## AMS
+
+AMS is the micro-service in OCM which hold the users clusters list, which is indeed a list of `Subscriptions` AMS objects, and handles the authZ for assisted-service API calles.
+Somethimes when people refers to OCM they actually mean AMS inside OCM.
+
+## Cluster lifecycle
+
+### Cluster registration
+
+On cluster registration, the service will create an AMS subscription for the cluster which some initial values:
+```
+status: "Reserved"
+cluster_id: The cluster id registered in AI DB.
+display_name: The cluster name in AI DB.
+product_id: "OCP"
+product_category: "AssistedInstall"
+```
+
+### Cluster is updated with a new cluster-name
+
+In this case the service will patch the subscription with the new cluster name.
+```
+display_name: New cluster name
+```
+
+### Cluster installation
+
+During the cluster installation process we will have multiples point of integration with AMS.
+
+Once an `openshift_cluster_id` is generated during the `preparing-for-installation` state, the service will patch the subscription with:
+```
+external_cluster_id: Openshift cluster ID
+```
+
+Later on, when console operator is installed during the `finalizing` state, the service will patch the subscription with:
+```
+console_url: Cluster's console URL
+```
+
+Finally, when the cluster is fully installed (or degraded), the service will patch the subscription with:
+```
+status: "Active"
+```
+
+### Clsuter is sending metrics to Openshift Telemeter
+
+Once metrircs reaches Telemeter server, Telemeter will notify AMS and AMS will seach for a subscription with a matching `external_cluster_id` (it is included in the metrics the cluster send). If it finds such subscription it will add a `metrics` field to the subscription, otherwise, it will create a new subscription for that "unsubscribed" cluster but in this case all the data patched by AI will be missing - this should be a bug indication. 
```

Comment:
```suggestion
Once metrics reach the Telemeter server, Telemeter will notify AMS and AMS will search for a subscription with a matching `external_cluster_id` (which is included in the metrics the cluster send). If it finds such a subscription it will add a `metrics` field to the subscription, otherwise, it will create a new subscription for that "unsubscribed" cluster but in this case, all the data patched by AI will be missing - this should be a bug indication. 
```